src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -7459,7 +7465,7 @@ private BoundExpression BadIndexerExpression(ExpressionSyntax node, BoundExpress
         }
 
         private BoundExpression BindElementAccessCore(
-             ExpressionSyntax node,
+             SyntaxNode node,
```

Comment:
Should we assert anything about these notes? Must be expression or pattern, for example?

Reply:
> Should we assert anything about these nodes? Must be expression or pattern, for example?

In general we don't expect any particular syntax. This is just a reference to source to report errors and link bound nodes to.